modules/xfem/test/tests/bimaterials/glued_bimaterials_2d.i

```diff
@@ -0,0 +1,298 @@
+[GlobalParams]
+  order = FIRST
+  family = LAGRANGE
+  displacements = 'disp_x disp_y'
+[]
+
+[XFEM]
+  qrule = volfrac
+  output_cut_plane = true
+[]
+
+[UserObjects]
+  [./level_set_cut_uo]
+    type = LevelSetCutUserObject
+    level_set_var = ls
+  [../]
+[]
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 5
+  ny = 5
+  xmin = 0.0
+  xmax = 5.
+  ymin = 0.0
+  ymax = 5.
+  elem_type = QUAD4
+  displacements = 'disp_x disp_y'
+[]
+
+[MeshModifiers]
+  [./left_bottom]
+    type = AddExtraNodeset
+    new_boundary = 'left_bottom'
+    coord = '0.0 0.0'
+  [../]
+  [./left_top]
+    type = AddExtraNodeset
+    new_boundary = 'left_top'
+    coord = '0.0 5.'
+  [../]
+[]
+
+[AuxVariables]
+  [./ls]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[AuxKernels]
+  [./ls_function]
+    type = FunctionAux
+    variable = ls
+    function = ls_func
+  [../]
+[]
+
+[Variables]
+  [./disp_x]
+  [../]
+  [./disp_y]
+  [../]
+[]
+
+[Functions]
+  [./ls_func]
+    type = ParsedFunction
+    value = 'y-2.5'
+  [../]
+[]
+
+[AuxVariables]
+  [./stress_xx]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./stress_yy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./stress_xy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./a_strain_xx]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./a_strain_yy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./a_strain_xy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./b_strain_xx]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./b_strain_yy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./b_strain_xy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[Kernels]
+  [./TensorMechanics]
+  [../]
+[]
```

Comment:
Why do you still use the old tensor mechanics legacy action in this input file? Could you use the tensor mechanics action instead?

Reply:
@sapitts Do you know how to use the master actions for materials with base_names? I need to calculate A_strain, B_strain, A_elasticity, B_elasticity, A_stress, B_stress, etc.

Reply:
I didn't find an input file example, but the master action does have the base_name parameter: https://github.com/idaholab/moose/blob/devel/modules/tensor_mechanics/src/actions/TensorMechanicsActionBase.C#L51

Reply:
I just quick scanned the codes and I think it can only add single strain/stress with a base_name, but not two different base_names. 

I guess we could work this out in the future.